### ISSUE TITLE
My forked github pages didnt show anything, just white blank page

### ISSUE THREAD

**Issue by benagastov** [ztjhz/FreeChatGPT#74]
I already ensured that I deploy pages with main branch in action workflow tab at first time. But then I changed the branch in pages setting into gh-pages and yet github pages just show white blank page. I check workflow tab and its still has green mark, no error

**Comment by ztjhz** [comment]
When using a subpage, please ensure to set the correct `base` in `vite.config.js`. By default, `base` is set to `/`, so you'll need to update it for subpages. For example, if you're deploying to `https://<USERNAME>.github.io/<REPO>`, set the `base` to `/REPO/`.

To make things easier, here's the code you can use in your `vite.config.js` file:

```js
export default defineConfig({
  base: '/REPO/'
.
.
.
```

We'll also update the deployment instructions in the readme to reflect this change.

**Comment by benagastov** [comment]
Oh you mean I cannot use Github Pages directly? I never use Javascript or React, I just learn C, Matlab and Python. But I think it is good idea if you can add some details specific for your repo so I can try using some basic tutorials from other website and books for beginner.

**Comment by ztjhz** [comment]
Yes, you can definitely use GitHub Pages to host your website. However, you will need to configure your `vite.config.js` file since GitHub Pages is hosted at a subpage. The URL for GitHub pages follows this format: `https://<USERNAME>.github.io/<REPO>`. This is different from the main page URL, which is `https://<USERNAME>.github.io`.

If you are using a custom domain name, then there is no further configuration needed on the GitHub Pages side. However, you may still need to configure your DNS settings for your custom domain name.

**Comment by Mxililoves** [comment]
I had the same issue after creating another template from a working project.
To fix it I had to change Workflow permission through Repo -> Settings -> Actions -> General
Set:
Read and write permissions
Check:
Allow Github Actions to create and approve pull requests

**Comment by benagastov** [comment]
> Yes, you can definitely use GitHub Pages to host your website. However, you will need to configure your `vite.config.js` file since GitHub Pages is hosted at a subpage. The URL for GitHub pages follows this format: `https://<USERNAME>.github.io/<REPO>`. This is different from the main page URL, which is `https://<USERNAME>.github.io`.
> 
> If you are using a custom domain name, then there is no further configuration needed on the GitHub Pages side. However, you may still need to configure your DNS settings for your custom domain name.

Thank you but where is the file? I cannot find it

**Comment by benagastov** [comment]
> Yes, you can definitely use GitHub Pages to host your website. However, you will need to configure your `vite.config.js` file since GitHub Pages is hosted at a subpage. The URL for GitHub pages follows this format: `https://<USERNAME>.github.io/<REPO>`. This is different from the main page URL, which is `https://<USERNAME>.github.io`.
> 
> If you are using a custom domain name, then there is no further configuration needed on the GitHub Pages side. However, you may still need to configure your DNS settings for your custom domain name.

Man You Are the BEST! I copy the code from config.js and then ask ChatGpt to explain it. And then I copy all your replies and then ChatGPT fix the config code. Amazing amazing
